session/session.go

```diff
@@ -1340,6 +1340,13 @@ func (s *session) SetGlobalSysVar(name, value string) (err error) {
 	if value, err = sv.Validate(s.sessionVars, value, variable.ScopeGlobal); err != nil {
 		return err
 	}
+	if name == variable.MaxAllowedPacket {
+		u, err := variable.TruncateMaxAllowedPacket(s.sessionVars.StmtCtx, value)
+		if err != nil {
+			return err
+		}
+		value = strconv.FormatUint(u, 10)
+	}
```

Comment:
Can this be moved to the validation func? The validation func should be able to check if Scope == ScopeGlobal.